activation/wire/wire_v2_helpers.go

```diff
@@ -0,0 +1,114 @@
+package wire
+
+import (
+	"math/rand/v2"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/signing"
+)
+
+type testAtxV2Opt func(*ActivationTxV2)
+
+func WithMarriageCertificate(sig *signing.EdSigner, refAtx types.ATXID, atxPublisher types.NodeID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		certificate := MarriageCertificate{
+			ReferenceAtx: refAtx,
+			Signature:    sig.Sign(signing.MARRIAGE, atxPublisher.Bytes()),
+		}
+		atx.Marriages = append(atx.Marriages, certificate)
+	}
+}
+
+func WithMarriageATX(id types.ATXID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		atx.MarriageATX = &id
+	}
+}
+
+func WithPublishEpoch(epoch types.EpochID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		atx.PublishEpoch = epoch
+	}
+}
+
+func WithInitial(commitAtx types.ATXID, post PostV1) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		atx.Initial = &InitialAtxPartsV2{
+			CommitmentATX: commitAtx,
+			Post:          post,
+		}
+	}
+}
+
+func WithPreviousATXs(atxs ...types.ATXID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		atx.PreviousATXs = atxs
+	}
+}
+
+func WithNIPost(opts ...testNIPostV2Opt) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		nipost := &NIPostV2{}
+		for _, opt := range opts {
+			opt(nipost)
+		}
+		atx.NIPosts = append(atx.NIPosts, *nipost)
+	}
+}
+
+type testNIPostV2Opt func(*NIPostV2)
+
+func WithNIPostChallenge(challenge types.Hash32) testNIPostV2Opt {
+	return func(nipost *NIPostV2) {
+		nipost.Challenge = challenge
+	}
+}
+
+func WithNIPostMembershipProof(proof MerkleProofV2) testNIPostV2Opt {
+	return func(nipost *NIPostV2) {
+		nipost.Membership = proof
+	}
+}
+
+func WithNIPostSubPost(subPost SubPostV2) testNIPostV2Opt {
+	return func(nipost *NIPostV2) {
+		nipost.Posts = append(nipost.Posts, subPost)
+	}
+}
+
+// NewTestActivationTxV2 creates a new ActivationTxV2 with random values.
+// ONLY FOR TESTING.
+func NewTestActivationTxV2(opts ...testAtxV2Opt) *ActivationTxV2 {
```

Comment:
I find taking a `*testing.T` as the first parameter a nice way to enforce that a function is used only for tests. How about:
```suggestion
// NewActivationTxV2 creates a new ActivationTxV2 with random values.
func NewActivationTxV2(_ *testing.T, opts ...testAtxV2Opt) *ActivationTxV2 {
```